src/tests/system/tests/test_gpo.py

```diff
@@ -0,0 +1,1056 @@
+           SeInteractiveLogonRight key.
+        4. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing'
+        5. Start SSSD
+    :steps:
+        1. Authenticate as 'user1' and 'user2'
+    :expectedresults:
+        1. Users authentication is successful
+    :customerscenario: True
+    """
+    user1 = ad.user("user1").add()
+    user2 = ad.user("user2").add()
+    group = ad.group("group").add().add_members([user2])
+
+    ad.gpo("site policy").add().policy(
+        {
+            "SeInteractiveLogonRight": [user1, group, ad.group("Domain Admins")],
+            "SeDenyInteractiveLogonRight": [],
+        }
+    ).link()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.start()
+
+    assert client.auth.parametrize(method).password(username="user1", password="Secret123")
+    assert client.auth.parametrize(method).password(username="user2", password="Secret123")
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["su", "ssh"])
+@pytest.mark.topology(KnownTopology.AD)
+def test_gpo__is_set_to_enforcing_and_users_are_denied(client: Client, ad: AD, method: str):
+    """
+    :title: Group policy object host base access control is set to enforcing and users are denied
+    :description:
+        When ad_gpo_access_control is set to enforcing, user access is managed and enforced by the policy.
+        Unlisted users are denied access.
+    :setup:
+        1. Create the following users; 'user1', 'deny_user1', 'deny_user2'
+        2. Create the 'deny_group' and add 'deny_user2' as a member
+        3. Create and link the GPO 'site policy' and add 'Domain Admins' to SeInteractiveLogonRight
+           key. Add 'deny_user1' and 'deny_group' to SeDenyInteractiveLogonRight key
+        4. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing'
+        5. Start SSSD
+    :steps:
+        1. Authenticate as 'deny_user1' and 'deny_user2'
+        2. Check logs for gpo_access_check result
+    :expectedresults:
+        1. Users authentication is unsuccessful
+        2. Access check result is denied
+    :customerscenario: True
+    """
+    deny_user1 = ad.user("deny_user1").add()
+    deny_user2 = ad.user("deny_user2").add()
+    deny_group = ad.group("deny_group").add().add_members([deny_user2])
+
+    ad.gpo("site policy").add().policy(
+        {
+            "SeInteractiveLogonRight": [ad.group("Domain Admins")],
+            "SeDenyInteractiveLogonRight": [deny_user1, deny_group],
+        }
+    ).link()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.start()
+
+    assert not client.auth.parametrize(method).password(username="deny_user1", password="Secret123")
+    assert not client.auth.parametrize(method).password(username="deny_user2", password="Secret123")
+
+    log_str = client.fs.read(client.sssd.logs.domain())
+    assert "access_denied = 1" in log_str
```

Comment:
Imo if logs checking is used at all, it should be more specific here. If you can do the equivalent of `grep -E '(user_sid|access_denied|access_gr)'`  you would get block of  three lines like e.g.
```
(2024-04-19 18:18:10): [be[samba.test]] [ad_gpo_access_check] (0x0400): [RID#8]        user_sid = S-1-5-21-1273581344-3761274449-2740015630-500
(2024-04-19 18:18:10): [be[samba.test]] [ad_gpo_access_check] (0x0400): [RID#8]  access_granted = 1
(2024-04-19 18:18:10): [be[samba.test]] [ad_gpo_access_check] (0x0400): [RID#8]   access_denied = 0
```
which can be properly checked for each login attempt individually.

Reply:
ACK, will change this.

Reply:
Thinking about it some more, I don't think log checking is necessary here. I do want to improve the test framework for other tests; I'll add a TODO for later, though. Are you okay with no log checking for this test?